public/js/reports/filter-prompt.component.js

```diff
@@ -82,15 +90,21 @@
         function getFilterValues (term) {
             const options = {
                 contains: term,
-                limit: 15,
             };
 
-            return api.getReportFilterValues(vm.filter, options).then(function (result) {
-                const values = result.data.map(row => row.f).filter(f => f !== null && f !== undefined && f.trim() !== '');
-                vm.values = values;
-
-                return values;
-            });
+            if (vm.filter.suggestions.show) {
+                if (vm.filter.suggestions.limit === 0 || typeof (vm.filter.suggestions.limit) !== 'number' || Math.sign(vm.filter.suggestions.limit) === -1) {
+                    options.limit = 15;
+                } else {
+                    options.limit = vm.filter.suggestions.unlimited ? null : parseInt(vm.filter.suggestions.limit, 10);
+                }
```

Comment:
It can be written in a more simple way:
```js
const limit = parseInt(vm.filter.suggestions.limit, 10) || 15;
if (!vm.filter.suggestions.unlimited && limit > 0) {
    options.limit = limit;
}
```

Reply:
thanks ! you're right I change it